src/utils/__test__/DatasetUtils.spec.js

```diff
@@ -81,3 +81,50 @@ describe('datasets tree', () => {
     expect(tree).toStrictEqual(result);
   });
 });
+
+const datasetObjectForFromScratch = {
+  name: 'Dataset1',
+  description: undefined,
+  tags: undefined,
+  sourceType: 'None',
+};
+
+const filteredDatasetObjectForFromScratch = {
+  name: 'Dataset1',
+  sourceType: 'None',
+};
+
+const datasetObjectForADT = {
+  name: 'Dataset1',
+  description: undefined,
+  tags: undefined,
+  sourceType: 'ADT',
+  source: {
+    path: undefined,
+    location: 'location',
+    name: undefined,
+  },
+};
+
+const filteredDatasetObjectForADT = {
+  name: 'Dataset1',
+  sourceType: 'ADT',
+  source: {
+    location: 'location',
+  },
+};
+
+describe('datasets object', () => {
+  test.each`
+    initialDatasetObject           | filteredDatasetObject
+    ${datasetObjectForFromScratch} | ${filteredDatasetObjectForFromScratch}
+    ${datasetObjectForADT}         | ${filteredDatasetObjectForADT}
+    ${{}}                          | ${{}}
+    ${null}                        | ${null}
+    ${undefined}                   | ${undefined}
+    ${[]}                          | ${[]}
```

Comment:
Maybe add "extreme cases" with number, string, boolean ?

Reply:
I don't know if it's useful because we already know types returned by RHF, @csm-thu your opinion ? :slightly_smiling_face:

Reply:
I agree, if RHF always returns an object, then we don't need to add cases for other POD types :+1: 
(we must take care not to use `removeUndefinedValuesBeforeCreatingDataset` from another context though)